### ISSUE TITLE
Give more priority to museums

### ISSUE THREAD

**Issue by dieterdreist** [gravitystorm/openstreetmap-carto#3924]
### Expected behavior
they outrule bars and banks
### Actual behavior
the get hidden because of bars and banks
### Links and screenshots illustrating the problem
Here's an example https://www.openstreetmap.org/node/6205724716
I am not completely sure whether there are other sideeffects, but it seems as if the museum is hidden because of the bank and the bar. Please check the priorities.

**Comment by HolgerJeromin** [comment]
Current screenshot shows exactly what you suggest in all zoom levels:
![image](https://user-images.githubusercontent.com/2410353/66314855-f923c700-e914-11e9-8b4e-c086f6862dc4.png)

**Comment by dieterdreist** [comment]
great to read it, but it doesn't represent the situation on my screen (and the museum was added 9 months ago and is still in version 1):
![Screenshot 2019-10-07 at 15 42 13](https://user-images.githubusercontent.com/581288/66317077-24101a00-e919-11e9-91de-4577c0c1b5a8.png)
![Screenshot 2019-10-07 at 15 42 07](https://user-images.githubusercontent.com/581288/66317086-270b0a80-e919-11e9-8f37-0a9f19cdde10.png)
![Screenshot 2019-10-07 at 15 41 54](https://user-images.githubusercontent.com/581288/66317093-2a05fb00-e919-11e9-8eae-bcbde2231907.png)
![Screenshot 2019-10-07 at 15 41 45](https://user-images.githubusercontent.com/581288/66317101-2d00eb80-e919-11e9-84bb-818b3400e87e.png)

**Comment by jeisenbe** [comment]
This is the current line where `tourism=museum` is selected. It is before all `amenity` features, so museums should have priority over banks, bars, etc:

https://github.com/gravitystorm/openstreetmap-carto/blob/ea73a44105ad1c3a0ebff817d4a8e9ea9233124b/project.mml#L1413-L1418

Following the link, I get the same results as @HolgerJeromin - https://www.openstreetmap.org/node/6205724716

So this is a problem with one of the rendering servers, not with this style. 

(There is a problem on z18 where the museum icon is shown, but the bar text shows: the museum name is too long to fit in the space, so it is not rendered, but then the bar name fits, so it is shown. This is a known issue, see #234 and PR #2597 which attempted to solve it https://www.openstreetmap.org/node/6205724716#map=18/44.38999/7.54744)

**Comment by jeisenbe** [comment]
I suppose we could consider giving museums priority over `aeroway=helipad`, `tourism=motel/apartment/etc` - is this also an issue?

**Comment by matkoniecz** [comment]
> This is the current line where tourism=museum is selected. It is before all amenity features, so museums should have priority over banks, bars, etc:

AFAIK order in COALESCE matters when multiple things are tagged at once on one object. So `tourism=museum military=bunker` will get `feature=tourism_museum`, not `feature=military_bunker`.

Compare with say https://github.com/gravitystorm/openstreetmap-carto/blob/ea73a44105ad1c3a0ebff817d4a8e9ea9233124b/project.mml#L1630-L1635 where objects are ordered by type for rendering order

**Comment by jeisenbe** [comment]
Related to #3880 - Symbol and label prioritization is not in sync with the starting zoom levels - probably would be fixed if that issue is solved.

**Comment by StyXman** [comment]
> order in `COALESCE` matters when multiple things are tagged at once on one object. So `tourism=museum military=bunker` will get `feature=tourism_museum`, not `feature=military_bunker`

Agreed, but the original question was, as I understand it, when a museum is close to other amenities (POIs?), render it "with higher priority" than those other amenities.

**Comment by matkoniecz** [comment]
Yes, I just wanted to explain when this specific order matters.

**Comment by dieterdreist** [comment]
FWIW, no matter how often I clear the browser cache I still see the situation like in the screenshots I have provided. If people see different map tiles depending on the server from which they get them, this would be a different issue not concerning this repo, or would it?

**Comment by matkoniecz** [comment]
> or would it?

Looking at code it seems that there is no well defined order of objects, at least in this case. I think that adding something along ORDER_BY osm_id to ensure that all servers serve the same tiles would be a good idea.

**Comment by pnorman** [comment]
> FWIW, no matter how often I clear the browser cache I still see the situation like in the screenshots I have provided. If people see different map tiles depending on the server from which they get them, this would be a different issue not concerning this repo, or would it?

If the ordering of objects is not deterministic then it's a style bug.
